### PR TITLE
Install hostmetrics app configuration (opt-in)

### PR DESCRIPTION
**Pull request by portertech**
This pull-request provides an opt-in installation flag/option (`--install-hostmetrics`) to fetch and install the hostmetrics configuration (`examples/conf.d/${platform}.yaml`). This currently works on Linux and MacOS (darwin).

The installation command:

```
curl -s https://raw.githubusercontent.com/SumoLogic/sumologic-otel-collector/main/scripts/install.sh | sudo -E bash -s -- --installation-token "${SUMOLOGIC_INSTALL_TOKEN}"
```

Becomes (adding `--install-hostmetrics`):

```
curl -s https://raw.githubusercontent.com/SumoLogic/sumologic-otel-collector/main/scripts/install.sh | sudo -E bash -s -- --installation-token "${SUMOLOGIC_INSTALL_TOKEN}" --install-hostmetrics
```

This results in creating a `conf.d` configuration file:

```
/etc/otelcol-sumo/conf.d/hostmetrics.yaml
```

### CODE REVIEW

**Comment by portertech** [comment]
Test is going to fail until `examples/conf.d/${platform}.yaml` hits the main branch.

File: examples/conf.d/darwin.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+service:
+  pipelines:
+    metrics/mac/localhost:
+      receivers:
+        - hostmetrics/mac/localhost
+      processors:
+        - memory_limiter
+        - sumologic_schema
+        - resource/mac_resource_attributes/localhost
+        - resourcedetection/system
```
Comment (swiatekm-sumo) [inline examples/conf.d/darwin.yaml:30]:
```suggestion
        - resource/mac_resource_attributes/localhost
        - resourcedetection/system
        - sumologic_schema
```
As a rule of thumb, the sumo schema processor should be close to the end of the pipeline. It changes metadata from OT schema to Sumo schema, and we want to operate on the OT schema as much as we can.

File: examples/conf.d/linux.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+service:
+  pipelines:
+    metrics/linux/localhost:
+      receivers:
+        - hostmetrics/linux/localhost
+      processors:
+        - memory_limiter
+        - sumologic_schema
+        - resource/linux_resource_attributes/localhost
+        - resourcedetection/system
```
Comment (swiatekm-sumo) [inline examples/conf.d/linux.yaml:48]:
```suggestion
        - resource/linux_resource_attributes/localhost
        - resourcedetection/system
        - sumologic_schema
```

File: examples/conf.d/linux.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      exporters:
+        - sumologic
+    logs/linux/localhost:
+      receivers:
+        - filelog/linux/localhost
+      processors:
+        - memory_limiter
+        - sumologic_schema
+        - resource/linux_resource_attributes/localhost
+        - resourcedetection/system
```
Comment (swiatekm-sumo) [inline examples/conf.d/linux.yaml:59]:
```suggestion
        - resource/linux_resource_attributes/localhost
        - resourcedetection/system
        - sumologic_schema
```

File: pkg/scripts_test/check.go [hunk 4]
```diff
@@ -111,6 +111,14 @@ func checkEnvTokenInConfig(c check) {
 	require.Equal(c.test, token, conf.Extensions.Sumologic.InstallToken, "install token is different than expected")
 }
 
+func checkHostmetricsConfigCreated(c check) {
+	require.FileExists(c.test, hostmetricsConfigPath, "hostmetrics configuration has not been created properly")
```
Comment (swiatekm-sumo) [inline pkg/scripts_test/check.go:115]: Can you also check permissions and ownership here? We've had a bunch of issues related to this, and I want to ensure correctness for any new additions.

File: scripts/install.sh [hunk 5]
```diff
@@ -542,6 +551,15 @@ function setup_config() {
         exit 1
     fi
 
+    if [[ "${INSTALL_HOSTMETRICS}" == "true" ]]; then
+        echo -e "Installing ${OS_TYPE} hostmetrics configuration"
+        HOSTMETRICS_CONFIG_URL="https://raw.githubusercontent.com/SumoLogic/sumologic-otel-collector/${CONFIG_BRANCH}/examples/conf.d/${OS_TYPE}.yaml"
+        if ! curl --retry 5 --connect-timeout 5 --max-time 30 --retry-delay 0 --retry-max-time 150 -f -s "${HOSTMETRICS_CONFIG_URL}" -o "${CONFIG_DIRECTORY}/conf.d/hostmetrics.yaml"; then
+            echo "Cannot obtain hostmetrics configuration for '${CONFIG_BRANCH}' branch"
+            exit 1
```
Comment (swiatekm-sumo) [inline scripts/install.sh:559]: I wouldn't mind putting this in a separate function, to keep the main script body cleaner.

**Pull request merged**